python/torch_mlir_e2e_test/torchscript/framework.py

```diff
@@ -332,7 +333,9 @@ def run_workers_in_parallel(task_queue: mp.Queue, worker):
 
 def run_tests(tests: List[Test], config: TestConfig, sequential = False) -> List[TestResult]:
     """Invoke the given `Test`'s with the provided `TestConfig`."""
-    if sequential:
+    # Disable OSX multithreading until is fixed
+    # https://github.com/llvm/torch-mlir/issues/994
+    if sequential or platform.system() == 'Darwin':
```

Comment:
It looks like this is a common problem from Python 3.8+ on MacOS when Apple swapped to using spawn instead of fork here ([reference](https://github.com/pytest-dev/pytest-flask/issues/104#issuecomment-577908228)). There's two other solutions:

1. Run `mp.set_start_method('fork')` at init time.
2. Explicitly use fork with `Process`: (line 325) `mp.get_context('fork').Process(...)` ([example in the wild](https://github.com/pytorch/pytorch/pull/36542/files#diff-353ea690c435cf81f51b9dde1494eeec64f99a7d342509ecd891094398974784R2217))

The latter seems like it's a better fit for us.

If we do it this way, we retain the ability to operate in parallel and I don't think we actually need to check for a platform since Windows is the only system that would break with `fork` as the mp default, and we don't support it.

Caveat emptor: I haven't tried this in any kind of exhaustive test whatsoever.

Reply:
great suggestion. works like a charm. pushed up a fix. PTAL.